proxy/Main.cc

```diff
@@ -392,14 +395,15 @@ class MemoryLimit : public Continuation
       delete this;
       return EVENT_DONE;
     }
-    if (_memory_limit == 0) {
-      // first time it has been run
-      _memory_limit = REC_ConfigReadInteger("proxy.config.memory.max_usage");
-      _memory_limit = _memory_limit >> 10; // divide by 1024
-    }
-    if (_memory_limit > 0) {
-      if (getrusage(RUSAGE_SELF, &_usage) == 0) {
-        Debug("server", "memory usage - ru_maxrss: %ld memory limit: %" PRId64, _usage.ru_maxrss, _memory_limit);
+
+    // "reload" the setting, we don't do this often so not expensive
+    _memory_limit = REC_ConfigReadInteger("proxy.config.memory.max_usage");
+    _memory_limit = _memory_limit >> 10; // divide by 1024
```

Comment:
Because...? Shouldn't this be documented to have a metric of 1 kilobyte? Speaking of which, where _is_ the documentation for this stuff?

Reply:
No, the metric is not in KB, but the output from getrusage() is. They (who ever implemented that old code) decided to divide the configuration by 1k.